### PR TITLE
Individual trace view

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Allows you to click into a trace to view details for it and all its child spans.

https://www.loom.com/share/de41a52bee0841ca9ce43d9a9a17f3bc

## How did you test this change?

Local click test. Can't test in prod because of the backend changes required to load the new UI.

## Are there any deployment considerations?

This is all only available to Highlight admins and only adding reads to the DB. We should monitor performance of querying spans for some of the larger traces.

As a follow up, we should also consider renaming things from `Trace` to `Span` where appropriate. E.g. our "Traces" page shows a list of spans, and when you click into that span it loads the trace for that span.

## Does this work require review from our design team?

Will have Julian look this over. I'm sure there will be some back and forth to refine the UI, but there will be a lot of iterating on this before it goes live.

### CODE REVIEW

File: frontend/src/pages/Traces/TracePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import { useNavigate } from 'react-router-dom'
+
+import { TracePage } from '@/pages/Traces/TracePage'
+import { useParams } from '@/util/react-router/useParams'
+
+import * as styles from './TracePanel.css'
+
+export const TracePanel: React.FC = () => {
+	const navigate = useNavigate()
+	const { project_id } = useParams<{
```
Comment (SpennyNDaJets) [inline frontend/src/pages/Traces/TracePanel.tsx:12]: Not sure if there is a preference for `useParams` vs 
`const { projectId } = useProjectId()`?

File: frontend/src/pages/Traces/utils.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	const hours = Math.floor(minutes / 60)
+	const days = Math.floor(hours / 24)
+
+	const dayString = days > 0 ? `${days}d` : ''
+	const hourString = hours > 0 ? `${hours % 24}h` : ''
+	const minuteString = minutes > 0 ? `${minutes % 60}m` : ''
+	const secondString = seconds > 0 ? `${seconds % 60}s` : ''
+	const millisecondString = milliseconds > 0 ? `${milliseconds % 1000}ms` : ''
+
+	if (dayString) {
```
Comment (SpennyNDaJets) [inline frontend/src/pages/Traces/utils.ts:48]: Nit: Looks good! Wonder if we can simplify the if statements?
```
[dayString, hourString, minuteString, secondString, millisecondString].join(" ").trim()
```

**Review by SpennyNDaJets — APPROVED** [review]
Click test looks great!

Just want to check that clicking elsewhere in the table/page should close the panel?

https://github.com/highlight/highlight/assets/17744174/29d8355d-6141-4c5c-b1e9-882b9d811e31

File: frontend/src/pages/Traces/TracePanel.tsx (hunk 1 above)
Comment (ccschmitz) [reply frontend/src/pages/Traces/TracePanel.tsx:12]: Yeah, should probably be `useProjectId` unless you need other params from the URL. Thanks for mentioning!

File: frontend/src/pages/Traces/utils.ts (hunk 2 above)
Comment (ccschmitz) [reply frontend/src/pages/Traces/utils.ts:48]: Yeah, good call!

Comment (ccschmitz) [reply frontend/src/pages/Traces/utils.ts:48]: Shoot, I don't think this quite accomplishes the same thing. I agree it's ugly/verbose, but I think it's fairly easy to reason about, so I'm going to leave it for now.

File: frontend/src/pages/Traces/TracePanel.tsx (hunk 1 above)
Comment (ccschmitz) [reply frontend/src/pages/Traces/TracePanel.tsx:12]: I think I'm going to merge as is and handle this in a follow up!

**Pull request merged**